src/request.ts

```diff
@@ -77,27 +77,59 @@ export async function buildRequestUrl(
 }
 
 /**
- * The error type returned by `executeRequestError`.
+ * The error type returned by the SDK whenever it makes a network request
+ * (internally, this happens in `executeRequest`). It implements, so is
+ * compatible with, `AxiosError`.
  */
-export class ExecuteRequestError extends Error {
-  originalError: AxiosError;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class ExecuteRequestError<T = any, D = any> extends Error implements AxiosError {
   responseStatus: number | null;
   responseMessage: string | null;
 
+  // Properties required by `AxiosError`.
+
+  config: AxiosRequestConfig<D>;
+  code?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  request?: any;
+  response?: AxiosResponse<T, D>;
+  isAxiosError: boolean;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  toJSON: () => object;
+
   /**
    * Creates an `ExecuteRequestError`.
    *
    * @param message - The error message.
-   * @param axiosError - The original axios error.
+   * @param axiosError - The original Axios error.
    * @param responseStatus - The response status, if found in the original error.
    * @param responseMessage - The response message, if found in the original error.
    */
   constructor(message: string, axiosError: AxiosError, responseStatus: number | null, responseMessage: string | null) {
+    // Include this check since ExecuteRequestError implements AxiosError, but
+    // we only expect original errors from Axios here.
+    if (axiosError instanceof ExecuteRequestError) {
+      throw new Error(
+        "Could not instantiate an `ExecuteRequestError` from an `ExecuteRequestError`, an original error from axios was expected"
+      );
+    }
+
+    // Set `Error` fields.
     super(message);
-    this.originalError = axiosError;
```

Comment:
I think you have to keep this and clearly document what happened. If we don't keep this then this update in and of itself is another breaking change to 3rd party devs who adapted to the previous breaking change, making this PR breaking again even though it's fixing a situation.

Reply:
I second this. There is no need to get rid of this field.

Reply:
Sure. The previous version that released this `v4.0.21` was rolled back and deprecated - but, I don't see the harm in keeping this here.